pkg/reconciler/clusterbuildtemplate/clusterbuildtemplate.go

```diff
@@ -107,14 +122,63 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 	logger := logging.FromContext(ctx)
 
 	// Get the ClusterBuildTemplate resource with this key
-	if _, err := c.clusterBuildTemplatesLister.Get(key); errors.IsNotFound(err) {
+	cbt, err := c.clusterBuildTemplatesLister.Get(key)
+	if errors.IsNotFound(err) {
 		// The ClusterBuildTemplate resource may no longer exist, in which case we stop processing.
 		logger.Errorf("clusterbuildtemplate %q in work queue no longer exists", key)
 		return nil
 	} else if err != nil {
 		return err
 	}
 
-	// TODO: Meaningful reconciliation.
+	if err := c.reconcileImageCaches(ctx, cbt); err != nil {
+		return err
+	}
+
 	return nil
 }
+
+func (c *Reconciler) reconcileImageCaches(ctx context.Context, cbt *v1alpha1.ClusterBuildTemplate) error {
```

Comment:
Can this be moved to a common location shared by both types of template reconcilers?

Reply:
I've made the parts that are common usable from the `buildtemplate` reconciler.